Collections/collections_home_task.py

```diff
@@ -0,0 +1,76 @@
+"""
+Home task for Module 2 - Collections
+
+Write a code, which will:
+
+1. create a list of random number of dicts (from 2 to 10)
+
+dict's random numbers of keys should be letter,
+dict's values should be a number (0-100),
+example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+2. get previously generated list of dicts and create one common dict:
+
+if dicts have same key, we will take max value, and rename key with dict number with max value
+if key is only in one dict - take it as is,
+example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+Each line of code should be commented with description.
+
+Commit script to git repository and provide link as home task result.
+"""
+import random
+
+# Create a list of random number of dicts (from 2 to 10)
+list_of_random_dictionaries = []  # Define empty list of dictionaries
+set_of_letters = (  # Define set of all alphabet letters
+    'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+    'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+    )
+for i in range(random.randint(2, 10)):  # FOR statement with randon number of iterations between 2 and 10
+    sample_dictionary = {random.choice(set_of_letters): random.randint(0, 100)
+                         for _ in range(random.randint(0, 100))}  # Generation of random dictionary
+    list_of_random_dictionaries.append(sample_dictionary)  # Adding of generated dictionary to list
+print(f'List of random dictionaries: {list_of_random_dictionaries}.')  # Printing of final list
+
+# Get previously generated list of dicts and create one common dict
+final_dict = {}  # Define empty final dictionary
+common_dict = {}  # Define empty temporary dictionary
+index = 0  # Default value for index of dictionaries in list
+max_key = ''  # Default value for max_key in all dictionaries
+try:  # Try for catch list index out of range error
+    for item in list_of_random_dictionaries:  # Take list of random dictionaries
```

Comment:
```suggestion
    for dictionary in list_of_random_dictionaries:  # Take list of random dictionaries
```
Try to make variable names more meaningful

Reply:
Fixed.